src/people.cpp

```diff
@@ -280,6 +350,61 @@ void Guests::NotifyRideDeletion(const RideInstance *ri) {
 	}
 }
 
+/** A guest complains that he is hungry and can't buy food. May send a message to the player. */
+void Guests::ComplainHunger()
+{
+	this->complaint_counter_hunger++;
+	if (this->time_since_complaint_hunger > COMPLAINT_TIMEOUT && this->complaint_counter_hunger >= COMPLAINT_THRESHOLD_HUNGER) {
+		this->complaint_counter_hunger = 0;
+		this->time_since_complaint_hunger = 0;
+		_inbox.SendMessage(new Message(GUI_MESSAGE_COMPLAIN_HUNGRY));
+	}
+}
+
+/** A guest complains that he is thirsty and can't buy a drink. May send a message to the player. */
+void Guests::ComplainThirst()
+{
+	this->complaint_counter_thirst++;
+	if (this->time_since_complaint_thirst > COMPLAINT_TIMEOUT && this->complaint_counter_thirst >= COMPLAINT_THRESHOLD_THIRST) {
+		this->complaint_counter_thirst = 0;
+		this->time_since_complaint_thirst = 0;
+		_inbox.SendMessage(new Message(GUI_MESSAGE_COMPLAIN_THIRSTY));
+	}
+}
+
+/** A guest complains that he needs a toilet and can't find one. May send a message to the player. */
+void Guests::ComplainWaste()
+{
+	this->complaint_counter_waste++;
+	if (this->time_since_complaint_waste > COMPLAINT_TIMEOUT && this->complaint_counter_waste >= COMPLAINT_THRESHOLD_WASTE) {
+		this->complaint_counter_waste = 0;
+		this->time_since_complaint_waste = 0;
+		_inbox.SendMessage(new Message(GUI_MESSAGE_COMPLAIN_TOILET));
+	}
+}
+
+/** A guest complains that the paths are very dirty. May send a message to the player. */
+void Guests::ComplainLitter()
+{
+	this->complaint_counter_litter++;
+	if (this->time_since_complaint_litter > COMPLAINT_TIMEOUT && this->complaint_counter_litter >= COMPLAINT_THRESHOLD_LITTER) {
+		this->complaint_counter_litter = 0;
+		this->time_since_complaint_litter = 0;
+		_inbox.SendMessage(new Message(GUI_MESSAGE_COMPLAIN_LITTER));
+	}
+}
+
+/** A guest complains that many path objects are demolished. May send a message to the player. */
+void Guests::ComplainVandalism()
+{
+	this->complaint_counter_vandalism++;
+	if (this->time_since_complaint_vandalism > COMPLAINT_TIMEOUT && this->complaint_counter_vandalism >= COMPLAINT_THRESHOLD_VANDALISM) {
+		this->complaint_counter_vandalism = 0;
+		this->time_since_complaint_vandalism = 0;
+		_inbox.SendMessage(new Message(GUI_MESSAGE_COMPLAIN_VANDALISM));
+	}
+}
```

Comment:
When you write code like this and use copy/paste (or are tempted to), stop coding, and think if there is a better solution to avoid having X times the same function in a program. It's almost never a good solution.